src/utils/spark.py

```diff
@@ -64,12 +65,37 @@ def __init__(self, app_name: str, password_file_path: str) -> None:
         self.initial_compression = "snappy"
         self.final_compression = "zstd"
 
-        # Store S3 details so workers can upload finished jobs
-        self.s3_client = boto3.client("s3")
-        self.s3_bucket = os.getenv("AWS_S3_BUCKET")
-        self.s3_prefix = os.getenv("AWS_S3_PREFIX", "iasworld")
-
+        # Create the Spark session and logging
         self.spark = SparkSession.builder.appName(self.app_name).getOrCreate()
+        self.log_file_path = f"/tmp/logs/{self.app_name}.log"
+        self.logger = self.get_logger()
+
+    def get_logger(self):
```

Comment:
This horror is extracting the log4j logger from the JVM and accessing it using a py4j wrapper. This lets us intersperse the Python logger output with the Spark log output from stdout. It also sets up a file sink to write to so that we have a file to upload to CloudWatch.

Reply:
**[Praise]** It's annoying that we have to dig into the PySpark internals to get access to this logger, but I think the end result of a logger that can be passed to other parts of the code to allow them to write to the Spark logs is a solid abstraction. Nice work!